File: March 27, 2026.js
function truncateText(str) {

  const width1 = "ilI";
  const width2 = "fjrt";
  const width3 = "abcdeghkmnopqrstuvwxyzJL";
  const width4 = "ABCDEFGHKMNOPQRSTUVWXYZ";
  let totalWidth = 0;
  let truncatedStr = "";

  for (let i = 0; i < str.length; i++) {

    if (width1.includes(str[i])) {
      totalWidth += 1;
    } else if (width2.includes(str[i])) {
      totalWidth += 2;
    } else if (width3.includes(str[i])) {
      totalWidth += 3;
    } else if (width4.includes(str[i])) {
      totalWidth += 4;
    } else if (str[i] === ".") {
      totalWidth += 1;
    } else if (str[i] === " ") {
      totalWidth += 2;
    }

    if (totalWidth > 47) {
      truncatedStr = str.slice(0, i) + "..."
      return truncatedStr;
    }

  }

  return str;
}

File: March 27, 2026.py
def truncate_text(s):

    width1 = "ilI"
    width2 = "fjrt"
    width3 = "abcdeghkmnopqrstuvwxyzJL"
    width4 = "ABCDEFGHKMNOPQRSTUVWXYZ"
    total_width = 0

    for i, char in enumerate(s):
        if char in width1:
            total_width += 1
        elif char in width2:
            total_width += 2
        elif char in width3:
            total_width += 3
        elif char in width4:
            total_width += 4
        elif char == ".":
            total_width += 1
        elif char == " ":
            total_width += 2

        if total_width > 47:
            truncated_s = s[:i] + "..."
            return truncated_s

    return s